### PR TITLE
Clothing Fabricator (WIP) 

### PR DESCRIPTION
**Pull request by SirBloodBuske**
Work In progess.

Clothing Fabricator 

A machine used to fabricate Clothing, and other objects for you to wear.

Could be hacked to get items like grey tide uniforms
 Emagging would give something traitor like, I suppose.

Good for the HoP or a shop.

Materials needed
 Plastics - Can get from chemistry and or other ways.
 Fabrics - can be made from cotton ( Botany ) or fur (Animals)
 (third material still WIP)

to make sure you won't be making thousands of uniforms it will cost a bit to make any one uniform, hacked or emagged uniforms will require more than normal and or special materials.

### CODE REVIEW

**Comment by SirBloodBuske** [comment]
fabnew

**Comment by SirBloodBuske** [comment]
fab3new

**Comment by SirBloodBuske** [comment]
Not sure what Items to add to this feature @Fox-McCloud

File: .gitignore [hunk 1]
```diff
@@ -15,3 +15,5 @@ data/
 /maps/RandomZLevels/backup/*
 /maps/cyberiad.dmm.backup
 /nano/debug.html
+maps/cyberiad.dmm
+paradise.dme
```
Comment (Iamgoofball) [inline .gitignore]: Hold the fuck up

File: code/game/machinery/fabricator.dm [hunk 2]
```diff
@@ -0,0 +1,569 @@
+#define FABRICATOR_MAIN_MENU       1
+#define FABRICATOR_CATEGORY_MENU   2
+#define FABRICATOR_SEARCH_MENU     3
+#define FABRICATOR 5
+/obj/machinery/fabricator
```
Comment (Iamgoofball) [inline code/game/machinery/fabricator.dm]: You copypasted the entirety of the autolathe code but renamed it. why didn't you just make this a child of the autolathe?

Comment (SirBloodBuske) [inline code/game/machinery/fabricator.dm]: Cause I am new to coding and I'm learning.

**Comment by ZomgPonies** [comment]
Not a huge fan of this. We already spawn clothes either in vendors or in lockers. I'd much rather see a machine to create cybernetic limbs that could then be used as the basis for a cybernetics lab.

**Comment by SirBloodBuske** [comment]
Well any input I like, If it is something we do not want then let me know other wise Ill keep on doing which to see what you would like!

**Comment by veganzombeh** [comment]
I'd agree with Ponies. All these things are already easily accessible, plus the average crewmember isn't going to have easy/legal access to glass and plastic to use this.

**Comment by ArgyleSweatshop** [comment]
It would make sense in the Head of Personnel's office

File: code/game/machinery/fabricator.dm (hunk 2 above)
Comment (Iamgoofball) [inline code/game/machinery/fabricator.dm]: Hey its all good. For the future though, a child of the autolathe would of worked a bit better

**Comment by SirBloodBuske** [comment]
I understand, But It is work in progress and I am taking all suggestions, The idea was it to be in a shop and or the HoP's office.

File: code/game/machinery/fabricator.dm (hunk 2 above)
Comment (Glloyd) [inline code/game/machinery/fabricator.dm]: Autolathe isn't that object oriented. Might be a good part of any future PRs doing the same thing.

**Comment by SirBloodBuske** [comment]
All that remains is finishing up the recipe list and what is needed to make said items, Also if glass will stay as a material or be changed to another.

**Comment by SamCroswell** [comment]
Unless I'm mistaken, 'Accessorys' should be 'Accessories'.

Also, I tend to agree with Ponies. We already have three varieties of vendors for various clothing items, I do not think we need any more sources of clothing.

**Comment by Fox-McCloud** [comment]
Yeah, going to have to echo @ZomgPonies  and @SamCroswell on this one--there's three vendors for clothing, a costume vendor, and a variety clothing vendor--not to mention a ton of closets, too.

**Pull request closed without merging**